types/node/fs.d.ts

```diff
@@ -536,70 +536,85 @@ declare module "fs" {
      * Asynchronous stat(2) - Get file status.
      * @param path A path to a file. If a URL is provided, it must use the `file:` protocol.
      */
-    export function stat(path: PathLike, options: BigIntOptions, callback: (err: NodeJS.ErrnoException | null, stats: BigIntStats) => void): void;
-    export function stat(path: PathLike, options: StatOptions, callback: (err: NodeJS.ErrnoException | null, stats: Stats | BigIntStats) => void): void;
     export function stat(path: PathLike, callback: (err: NodeJS.ErrnoException | null, stats: Stats) => void): void;
+    export function stat(path: PathLike, options: StatOptions & { bigint?: false } | undefined, callback: (err: NodeJS.ErrnoException | null, stats: Stats) => void): void;
+    export function stat(path: PathLike, options: StatOptions & { bigint: true }, callback: (err: NodeJS.ErrnoException | null, stats: BigIntStats) => void): void;
+    export function stat(path: PathLike, options: StatOptions | undefined, callback: (err: NodeJS.ErrnoException | null, stats: Stats | BigIntStats) => void): void;
```

Comment:
I think the `options === undefined` case is not optimal here as the result should be of type `Stats`.

Reply:
Its possible you could have a weakly defined `StatOptions | undefined` that you want to pass in, which should be legal. Since you can't differentiate between whether its a `Stats` or `BigIntStats` due to the weaker type, I would expect that to be the result. This definition looks fine to me.

Reply:
Also, the `undefined` case should hit the first overload (with `& { bigint?: false }`), which will correctly return `Stats`.